src/hdmf/utils.py

```diff
@@ -410,6 +410,9 @@ def fmt_docval_args(func, kwargs):
 
     Useful for methods that wrap other methods
     '''
+    warnings.warn("fmt_docval_args will be deprecated in a future version of HDMF. Please note that fmt_docval_args "
+                  "transparently removes all arguments not accepted by the function's docval.",
+                  PendingDeprecationWarning)
```

Comment:
Same comment as for below for ``call_docval_func``. I think many users/developers will be confused by the note about  . Instead, it would be useful to describe how to properly replace ``fmt_docval_args`` in code so folks know what to do.